### PR TITLE
[zh-cn]: add the translation of iterator overview

### PR DESCRIPTION
**Pull request by JasonLamv-t**
### Description

Same as the title.

### Motivation

Ditto.

### Additional details

Keys need to be considered:
- iterator helper
- proper iterator

### Related issues and pull requests

#17677

### CODE REVIEW

File: files/zh-cn/web/javascript/reference/global_objects/iterator/index.md [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+  next() {
+    // …
+  },
+});
+```
+
+## 构造函数
+
+- {{jsxref("Iterator/Iterator", "Iterator()")}} {{experimental_inline}}
+  - ：旨在被创建迭代器的其他类[继承](/zh-CN/docs/Web/JavaScript/Reference/Classes/extends)。直接构建会抛出错误。
```
Comment (jasonren0403) [inline files/zh-cn/web/javascript/reference/global_objects/iterator/index.md:124]:
```suggestion
  - ：旨在被创建迭代器的其他类[继承](/zh-CN/docs/Web/JavaScript/Reference/Classes/extends)。直接用于构建会抛出错误。
```

**Review by jasonren0403 — COMMENTED** [review]
一些其他的建议

File: files/zh-cn/web/javascript/reference/global_objects/iterator/index.md [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+- _Set 迭代器_，返回自 {{jsxref("Set.prototype.values()")}}、{{jsxref("Set.prototype.keys()")}}、{{jsxref("Set.prototype.entries()")}} 和 [`Set.prototype[@@iterator]()`](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Set/@@iterator)。
+- _正则表达式字符串迭代器_，返回自 [`RegExp.prototype[@@matchAll]()`](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/RegExp/@@matchAll) 和 {{jsxref("String.prototype.matchAll()")}}。
+- {{jsxref("Generator")}} 对象，返回自[生成器函数](/zh-CN/docs/Web/JavaScript/Reference/Statements/function*)。
+- _Segment 迭代器_，返回自 [`Intl.Segmenter.prototype.segment()`](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Intl/Segmenter/segment) 返回的 [`Segments`](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Intl/Segmenter/segment/Segments) 对象的 [`[@@iterator]()`](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Intl/Segmenter/segment/Segments/@@iterator) 方法。
+- _迭代器 Helper_，返回自迭代器辅助方法例如 {{jsxref("Iterator.prototype.filter()")}} 和 {{jsxref("Iterator.prototype.map()")}}。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/iterator/index.md:23]:
```suggestion
- _迭代器辅助方法_，返回自迭代器辅助方法例如 {{jsxref("Iterator.prototype.filter()")}} 和 {{jsxref("Iterator.prototype.map()")}}。
```

**Review by jasonren0403 — APPROVED** [review]
LGTM 👍

**Pull request merged**